File: sdk/LinguaNex.Js/src/linguanex.js
const localesData = {};
const axios = require('axios');
var linguaNexOptions = {};
// 从API获取国际化资源数据
function loadApiResources(prject, locale) {
    return axios.get(`${linguaNexOptions.baseUrl}/api/OpenApi/Resources/${prject}`, {
        headers: {
            "Accept-Language": locale
        }
    })
    .then(res => {
        localesData[locale] = res.data[0].resources
        return locale
    })
    .catch(err => {
        return new Error(err.message)
    })
    ;
  }
function setLocale(locale){
    if(!localesData[locale]){
        linguaNexOptions.currentLocale = locale;
        return loadApiResources(linguaNexOptions.project, locale, () => {});
    }
    else{
        linguaNexOptions.currentLocale = locale;
        return new Promise(s => s(null));
    }
}
// 获取国际化文本
function L(key, locale, defaultStr) {
    if(!locale)
        locale = linguaNexOptions.currentLocale
    const data = localesData[locale];
    if(data[key])
        return data[key];
    return defaultStr || key;
  }

function initLinguaNex(options) {
    linguaNexOptions = options;
    linguaNexOptions.currentLocale = options.defaultLocale;
    const promises = [];
    for(let locale in options.locales){
        promises.push(loadApiResources(options.project, options.locales[locale]));
    }
    return Promise.all(promises)
    .then(results => {
        console.log('Init Finish', results)
    })
    .catch(err => {
       console.error('Error:', err) 
    })
}
module.exports = {
    initLinguaNex: initLinguaNex,
    L: L,
    setLocale: setLocale
  };

File: sdk/LinguaNex.I18n.Go/linguanex.go
package linguanex

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

// LinguaNexOptions 定义了初始化LinguaNex所需的选项
type LinguaNexOptions struct {
	BaseUrl       string
	Project       string
	DefaultLocale string
	Locales       map[string]string
}

// localesData 存储所有本地化资源数据
var localesData = make(map[string]map[string]interface{})
var linguaNexOptions *LinguaNexOptions

// loadApiResources 从API获取国际化资源数据
func loadApiResources(project, locale string) (*http.Response, error) {
	req, err := http.NewRequest("GET", fmt.Sprintf("%s/api/OpenApi/Resources/%s", linguaNexOptions.BaseUrl, project), nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("Accept-Language", locale)

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}

	return resp, nil
}

// processApiResponse 处理API响应并填充localesData
func processApiResponse(locale string, resp *http.Response) error {
	defer resp.Body.Close()
	bodyBytes, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return err
	}

	var data []struct {
		Resources map[string]interface{} `json:"resources"`
	}
	err = json.Unmarshal(bodyBytes, &data)
	if err != nil {
		return err
	}

	if len(data) > 0 && data[0].Resources != nil {
		localesData[locale] = data[0].Resources
	} else {
		localesData[locale] = make(map[string]interface{})
	}

	return nil
}

// SetLocale 设置当前区域设置并加载资源
func SetLocale(locale string) error {
	if _, exists := localesData[locale]; !exists {
		linguaNexOptions.CurrentLocale = locale
		resp, err := loadApiResources(linguaNexOptions.Project, locale)
		if err != nil {
			return err
		}
		return processApiResponse(locale, resp)
	} else {
		linguaNexOptions.CurrentLocale = locale
		return nil
	}
}

// GetLocale 获取指定区域设置的资源
func GetLocale(locale string) map[string]interface{} {
	data, exists := localesData[locale]
	if exists {
		return data
	}
	return make(map[string]interface{})
}

// GetAllLocale 获取所有区域设置的资源
func GetAllLocale() map[string]map[string]interface{} {
	return localesData
}

// L 获取国际化文本
func L(key string, locale ...string) string {
	defaultStr := ""
	if len(locale) == 0 || locale[0] == "" {
		locale = []string{linguaNexOptions.CurrentLocale}
	}
	data := localesData[locale[0]]
	if data != nil && data[key] != nil {
		return data[key].(string)
	}
	return defaultStr
}

// Init 初始化LinguaNex实例
func Init(options LinguaNexOptions) error {
	linguaNexOptions = &options

	var promises []*http.Response
	for loc := range options.Locales {
		resp, err := loadApiResources(options.Project, loc)
		if err != nil {
			fmt.Println("Error:", err)
			continue
		}
		promises = append(promises, resp)
	}

	for _, resp := range promises {
		err := processApiResponse("", resp)
		if err != nil {
			fmt.Println("Error processing API response:", err)
		}
	}

	fmt.Println("Init Finish")
	return nil
}
